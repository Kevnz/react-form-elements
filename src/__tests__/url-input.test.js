import React from 'react'
import { render, cleanup, fireEvent } from 'react-testing-library'

import UrlInput from '../url-input'

describe('UrlInput component', () => {
  afterEach(() => {
    cleanup()
  })
  describe('The Rendered HTML', () => {
    it('should have a label', () => {
      const { getByText, container } = render(
        <UrlInput label="My UrlInput Label" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText('My UrlInput Label')
      expect(label.textContent).toBe('My UrlInput Label')
    })
  })
  describe('handling input', () => {
    it('should set a value when passed in but not be valid', () => {
      const { getByLabelText, container } = render(
        <UrlInput
          label="My Example URL"
          initialValue="http://www.example.com"
        />
      )

      expect(container.firstChild).toMatchSnapshot()
      const input = getByLabelText('My Example URL')
      expect(input.value).toBe('http://www.example.com')
      expect(input.validity.valid).toBe(true)
      fireEvent.change(input, { target: { value: 'Updated Value' } })
      expect(input.validity.valid).toBe(false)
    })
  })
})
