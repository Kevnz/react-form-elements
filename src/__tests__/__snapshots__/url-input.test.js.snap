// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlInput component The Rendered HTML should have a label 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-1-1"
  >
    My UrlInput Label
  </label>
  <input
    id="id-1-1"
    name="TextBox"
    type="url"
    value=""
  />
</div>
`;

exports[`UrlInput component handling input should set a value when passed in but not be valid 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-2-1"
  >
    My Example URL
  </label>
  <input
    id="id-2-1"
    name="TextBox"
    type="url"
    value="http://www.example.com"
  />
</div>
`;
